### DISCUSSION TITLE
Connection problems between Pc and Probat BRZ4

### DISCUSSION THREAD

**Discussion by ARCCoffeelab** [artisan-roaster-scope/artisan#1536] [Q&A]
### Problem Description

We bought a BRZ 4, installed the latest version of Artisan x Windows find on the website (2.8.4 BC Roasters) on our PC. 
we then also installed Phidget which detects our roaster and the temperature variation of the 4 thermocouples.
Following the instructions sent to us by Probat, we configured the machine as Probat UG Control and checked that the interface was the one indicated, but when activating the programme, it reports these errors:
- connection failed with S7
- the Websocket is disconnected or that the Phidget has not been started correctly and there is a Modbus error
- we subsequently modified the configuration as a Phidget, then tried with sample roaster....we also tried to modify parameters with the help of a computer expert but we did not get any results.

unfortunately we don't know how to proceed ...

### Artisan Version

Other - include in Description

### Computer OS and Version

Windows 10Pro

### Connected devices or roasting machine

Probat BRZ 4

### Files

- [ ] A sample profile *.alog file, renamed to *.alog.txt is attached.

- [ ] A settings file (Help>> Save Settings) *.aset, renamed to *.aset.txt is attached.

- [ ] Screenshots are attached.

- [X] No files are attached.

**Comment by MAKOMO** [comment]
> ### Problem Description
> We bought a BRZ 4, installed the latest version of Artisan x Windows find on the website (2.8.4 BC Roasters) on our PC. 

Well, 2.8.4 is three versions behind.  The latest version of Artisan is v2.10.4.

> we then also installed Phidget which detects our roaster and the temperature variation of the 4 thermocouples. 

Excellent!

> Following the instructions sent to us by Probat, we configured the machine as Probat UG Control and checked that the interface was the one indicated, but when activating the programme, 

Who at Probat told you configure Artisan for Probat UG Control? To my knowledge the BRZ 4 is not a Probat UGx and comes without any digital connector. As I understand you before, you installed 4 TCs and connected them to a Phidget. I assume you connect that Phidget to your PC, installed the Phidget driver and thus you need to configure Artisan to read that Phidget (whatever type it is).

> it reports these errors:
> 
> * connection failed with S7

Yes, because you configured Artisan for the Probat UG which communicates via the Siemens S7 protocol, but you do connect Phidgets and not a Siemens PLC.

> * the Websocket is disconnected or 

WebSockets are again a different protocol. Maybe you also configured for the Probat Series III or the (current) Probat Sample Roaster which both talk WebSockets!?

> that the Phidget has not been started correctly 

Maybe you did not configure for the correct type of Phidget module you connected?

> and there is a Modbus error

MODBUS is again a different protocol used by the older Probat Series II machines, and many other machines. Maybe you configured also for those.

> * we subsequently modified the configuration as a Phidget, then tried with sample roaster....we also tried to modify parameters with the help of a computer expert but we did not get any results.

You seem to be very lost. Try a Factory Reset (under menu Help) first to clean all those wired setups you combined somehow. Then choose the Phidget Module from the Meter popup under menu Config >> Devices. Maybe a VINT TMP1101 or the USB Phidget 1048?

> 
> unfortunately we don't know how to proceed ...
> 

Anyhow, you seem not to report a bug in the Artisan software but a question on how to configure it for your setup. I will move to the discussion board where it belongs and close this issue.

**Reply by ARCCoffeelab** [reply]
Where I can find the latest version v2.10.4. to download it?

We asked the Probat service department in Germany how to install/connect the roasting machine to the PC+ Artisan software, specifying that we bought the BRZ 4 in the digital version equipped with a USB connector but they sent us the instruction for Probat UG control....

We have connect the roaster machine to the PC with a USB cable, we installed the Phidget program... Where we finally see the roaster machine, the 4 thermocouples, etc....
We have already try with USB Phidget 1048...we tried with our IT technician every possible option.. but always the software say: connection failed with S7.

We will try now to download the right version... please tell us where to download it..and we will try to do what you suggested..
Thank you

**Reply by ARCCoffeelab** [reply]
Can you please tell us where we can find the latest version of the software so that we can download it?

Thank you

**Reply by MAKOMO** [reply]
Click "Install Now" on the [Artisan homepage](https://artisan-scope.org/). This always points you to the latest Artisan version.

Note that you need to do a "Factory Reset" first to get rid of this wrong S7 setup and clear every wrong setup you might have applied (not that settings are preserved on Artisan updates). I don't know the BRZ 4 roaster nor its "digital version". It sounds like Probat just adds a Phidget module and TCs to the machine. Anyhow, the BRZ 4 is not directly supported by Artisan. All setup support needs to be provided by Probat in this case who should now the machine and its "digital" variant best.

If Probat cannot solve the issue they might get into contact with the Artisan team to work this out. Not for free of course.

**Reply by ARCCoffeelab** [reply]
Can I have permission to quote what you sent me to Probat? to see what support they give us ?and what is their reply?

In the meantime, tomorrow I will do what you told me and see if something will change

Thank you

**Reply by MAKOMO** [reply]
Do what ever you like. Did they documented that their product comes with Artisan support?

**Reply by ARCCoffeelab** [reply]
[20221111_BRZ_PRE_PRG_Datasheet_EN.pdf](https://github.com/artisan-roaster-scope/artisan/files/15007429/20221111_BRZ_PRE_PRG_Datasheet_EN.pdf)

On the brochure of the roaster machine we bought:

quote (Pag.2)
"With the optional digital package, the sample roasters can be equipped with a digital temperature display and a USB
connection. This ensures the recording of temperature values and easy transmission to an external control system such as
Cropster or Artisan. "
uquote

Thank you for your support.
I will update you on my test results with your instructions.

**Reply by MAKOMO** [reply]
Got a call from Probat HQ. It seems those machines have a Phidget 1048 installed. To configure Artisan to read all 4 channels of a Phidget 1048 do the following:

- start Artisan, select menu Help >> Factory Reset to bring Artisan back to its initial configuration
- restart Artisan, select menu Config >> Machine >> Phidget >> "UBS 1048 Databridge 2xTC2 (this configures Artisan to read the first two channels as its ET and BT channels
- select menu Config >> Device, 2nd tab "Extra Devices", click Add, and set the "Device" of this first extra device to "Phidget 1048 4xTC 23" to read the other two channels from the 1048 module. Under "Label 1" and "Label 2" you can assign them some speaking names like "Barrel 3" and "Barrel 4" (note that you can also rename the ET/BT main channels under menu Config >> Curves, tab UI under "Rename ET and BT" to "Barrel 1" and "Barrel 2".
- save your settings via menu Help >> Save Settings to an `artisan-settings.aset` file as backup for quickly resetting those settings

**Reply by ARCCoffeelab** [reply]
Thank you very much.
I used your instructions and finally the roasting machine is connected to the PC and the software.
We finally solved the problem after two months.

**Reply by ARCCoffeelab** [reply]
Good Morning, we have again a problem with our BRZ4 and Artisan software.

If we use two roasters at the same time we can see the roasting curves of both but I can only report events for one roaster.
e.g we put coffee into the ROASTER 1 and during the roasting process we were able to make the software signal some events that were occurring - load, turning point, etc.., while we could not do this for the ROASTER 2 sample as once the "event" buttons were pressed it was no longer possible to do so....

we asked PROBAT questions, which I quote below:

- how can we roast in more than one roaster at the same time and have evidence of the succession of events on all the curves ?
-Is it possible to display one screen per roaster in order to have 'event' buttons on each one and manage the individual roasts individually ?
-(or) Is it possible to have all 4 roasting curves on one screen and be able to report events on all 4?
-how does the software understand if I have to use all the roasters at the same time in which I have already introduced coffee?  we did a test ... we brought roaster no. 3 and 4 up to temperature, introduced coffee into no. 3 pressed the load button but the software reported the event on no. 4.

NOW, THEIR ANSWER......

"quote "

-	As far as I know, it is not possible to map the events for all four roasters in one graph.
-	I am not aware that the software allows several separate screens or that you can start the software several times to achieve the desired result.
-	The software works with a temperature drop over time to realize the raw product has been filled. If this temperature drop is fast enough, an event is triggered. 
-	However, you can also create your own events, which then trigger events according to your wishes. These can be set under Config/Alarms.
-	I assume that this event was triggered for roaster 4 because the temperature was higher at the moment of filling. The software does not primarily assume that 4 roasters are mapped at the same time and the stored dependencies and conditions cannot then differ and are only based on the values. But since i don't know the exact dependencies of the software, this is just a guess.

"unquote"

we are a little bit confused ....
Hope you can help us
Thank you and Best regards

**Reply by mrpenner** [reply]
As they answered, Artisan is made to connect to one roaster. If you want to connect to more than one roaster at the same time, I think you'll need more devices, each running their own copy of Artisan.

If that doesn't clear it up, you might have to specify *what* you're confused about, if you want help with something.

**Reply by poundy** [reply]
the upshot here is that you have one instance of Artisan (that thinks it has one roaster connected) to one hardware set (the Phidget). Artisan only tracks one "roaster" or more specifically one "roast" even though you might have signals that come from different places (the additional barrels). It only knows to put data onto a graph, and everything gets treated as one roast. If you really want to treat each barrel as an individual roaster and track individual roasts, you'll need to "break apart" the equipment and software... That means a PC device running Artisan for each "roaster" and a phidget device connected to each PC device for each roaster (as I don't believe there's a way to multiplex them to give the data to each PC device from the one hardware platform). A second instance of Artisan on the same PC will only ever be able to run in "read only" mode, so you must have a separate machine to run each separate instance... 
Note I don't think any other commercial software would handle this scenario differently either, although I haven't ever tried.  It's just an unusual requirement to try to cater for.

**Reply by ARCCoffeelab** [reply]
@MAKOMO 
@mrpenner 
@poundy 

Thank you for your replies,we would like to send you a small giveaway from our company

**Reply by mrpenner** [reply]
I'm a fellow user trying to help where I can. You can contribute financially to Artisan development here: https://artisan-scope.org/donate/

**Reply by poundy** [reply]
agreed, I'm only another user who helps with support....
